File: 프로그래머스/3/42884. 단속카메라/단속카메라.py
def solution(routes):
    routes.sort(key = lambda x : x[1])
    ans = 1
    last_end = routes[0][1]
    for i in range(1, len(routes)):
        cur_start = routes[i][0]
        if last_end < cur_start:
            ans += 1
            last_end = routes[i][1]
        
    return ans

File: 프로그래머스/3/42884. 단속카메라/단속카메라.js
function solution(routes) {
    routes.sort((a, b) => a[1] - b[1]);
    let cnt = 1;
    let lastCamPos = routes[0][1];
    
    for (const route of routes) {
        const [start, end] = route;
        if (lastCamPos < start || lastCamPos > end) {
            lastCamPos = end;
            cnt += 1;
        }
    }
    
    return cnt;
}